Chapter7/call_count.erl

```diff
@@ -1,53 +1,69 @@
 -module(call_count).
--export([count/2]).
+-export([test/0]).
 -export([counter/1]).
 
+-define(COUNT_CALL(Module, Function), (counter ! {count, Module, Function} )).
+
 
 % Exercise 7-7: Counting Calls
-% How can you use the Erlang macro facility to count the number of calls to a particular function in a particular module?
 
-% No idea how to go about doing this? Does it want me to give it a file name of a file with erlang code
-% in it and then search the file for every call to a module? Or does it want me to make something that
-% checks how many times a module is called from the current module?
+test() ->
+  counter_start(),
+
+  foo(),
+  bar(),
+  foo(),
+  foo(),
+  bar(),
+
+  Result = counter_read(),
+  counter_stop(),
+  Result.
 
-% I just decided to make a program that given a file will count the ammount of words (Which can be "Module:Function")
 
-%%Example:
-% call_count:count("/Users/daniel.harris/Documents/GitErlang/learning-erlang/Chapter7/call_count.erl", "file:close").
+% Example Functions
+foo() ->
+  ?COUNT_CALL(call_count, foo),
```

Comment:
you could've used `?MODULE` and `?FUNCTION_NAME` instead

Reply:
Ok cool, I'll update the code now